src/core/enum_type.hpp

```diff
@@ -75,4 +75,61 @@ debug_inline constexpr void ToggleFlag(T &x, const T y)
 	}
 }
 
+template <typename Tenum, typename Tstorage>
+class EnumBitSet {
+public:
+	using enum_type = Tenum;
+	using storage_type = Tstorage;
+
+	constexpr EnumBitSet() : data(0) {}
+	constexpr EnumBitSet(Tenum other) : data(to_underlying(other)) {}
```

Comment:
```suggestion
	constexpr EnumBitSet(Tenum other) : data(1U << to_underlying(other)) {}
```
This should mimic `set`, right? Or maybe just call `set` in the constructor.

Or maybe the following, so you can add an arbitrary number of enums to the constructor.
```
	template <typename... T>
	constexpr EnumBitSet(T... others) : data(0)
	{
		(this->set(others), ...);
	}
```